### PR TITLE
[Notifier] Add Zendesk Notifier Bridge

### PR DESCRIPTION
**Pull request by stloyd**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| License       | MIT

Add Zendesk Notifier Bridge.

This bridge allows creating tickets in the Zendesk system via [Tickets API](https://developer.zendesk.com/api-reference/ticketing/tickets/tickets/)

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Zendesk/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CHANGELOG
+=========
+
+6.2
+---
+
+ * Added the bridge
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/CHANGELOG.md]:
```suggestion
 * Add the bridge
```

File: src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransport.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+{
+    private string $zendeskEmailAddress;
+    private string $zendeskApiToken;
+
+    public function __construct(
+        string $zendeskEmailAddress,
+        string $zendeskApiToken,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransport.php]: Please keep the actor signature on one line, thanks

File: src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransport.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Joseph Bielawski <stloyd@gmail.com>
+ */
+final class ZendeskTransport extends AbstractTransport
+{
+    private string $zendeskEmailAddress;
+    private string $zendeskApiToken;
+
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransport.php]:
```suggestion
    private string $email;
    private string $token;
```

would be enough

File: src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransportFactory.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        $host = $dsn->getHost();
+        if ('default' === $host) {
+            throw new IncompleteDsnException('Host is not set.', $dsn->getOriginalDsn());
+        }
+
+        if (str_ends_with($host, '.zendesk.com')) {
+            return $host;
+        }
+
+        return sprintf('%s.zendesk.com', $host);
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransportFactory.php]: Not sure we should add this "magic" here. I propose one must provide the full host ending with `.zendesk.com` without a fallback

File: src/Symfony/Component/Notifier/Bridge/Zendesk/composer.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=8.1",
+        "symfony/http-client": "^5.4|^6.0",
+        "symfony/notifier": "^5.4|^6.0"
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/composer.json]: The `UnsupportedSchemeException` in 5.4 does not know anything about this bridge, so it should need symfony/notifier 6.2

File: src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransportFactory.php (hunk 4 above)
Comment (stloyd) [reply src/Symfony/Component/Notifier/Bridge/Zendesk/ZendeskTransportFactory.php]: Replaced with above check + exception if doesn't contain that domain.

**Comment by stloyd** [comment]
@OskarStark @fabpot I guess this is ready for review, failures are not related to this PR.

File: src/Symfony/Component/Notifier/Bridge/Zendesk/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Zendesk Notifier
+===============
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/Zendesk/README.md]:
```suggestion
================
```

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Zendesk/README.md (hunk 6 above)
Comment (stloyd) [reply src/Symfony/Component/Notifier/Bridge/Zendesk/README.md]: Fixed 👍

**Comment by fabpot** [comment]
Thank you @stloyd.

**Comment by fabpot** [comment]
@stloyd Can you open a PR on symfony/recipes for the DSN?

**Comment by stloyd** [comment]
@fabpot Done: https://github.com/symfony/recipes/pull/1108

**Pull request merged**